modules/dnn/src/int8layers/quantization_utils.cpp

```diff
@@ -68,25 +150,65 @@ class QuantizeLayerImpl CV_FINAL : public QuantizeLayer
         CV_TRACE_FUNCTION();
         CV_TRACE_ARG_VALUE(name, "name", name.c_str());
 
-        CV_OCL_RUN(IS_DNN_OPENCL_TARGET(preferableTarget),
+        CV_OCL_RUN(IS_DNN_OPENCL_TARGET(preferableTarget) && !is1D,
                    forward_ocl(inputs_arr, outputs_arr, internals_arr))
 
         std::vector<Mat> inputs, outputs;
         inputs_arr.getMatVector(inputs);
         outputs_arr.getMatVector(outputs);
 
-        inputs[0].convertTo(outputs[0], CV_8S, 1.f/scale, zeropoint);
+        if (outputs[0].depth() != CV_8S)
+            outputs[0].convertTo(outputs[0], CV_8S);
+
+        if (is1D)
+        {
+            Mat inputTmp;
```

Comment:
Why do we need a temporary here?

Reply:
Thanks for your code review. I'm afraid in-place `convertTo` or `sub` or `div` will affect the original input data.

Reply:
What do you mean?

```cpp

divide(inputs[0], scalesMat, outputs[0]);
subtract(outputs[0], zeropointsMat, outpus[0]);
outputs[0].convertTo(outputs[0], CV_8S);

```

But I think the Mat will reallocate on `convertTo` even if there is enough space, so don't mind that.

Reply:
It won't affect anything afaict, it's just doesn't help at all in this case, so I'm marking this as resolved :)